### DISCUSSION TITLE
v3 RelationManager with headerActions, they dissapear

### DISCUSSION THREAD

**Discussion by Jam0r85** [filamentphp/filament#7200] [Help] [answered]
So i've just updated to v3 and everything is OK apart from this.

I have a PropertyResource and it has a TenancyRelationManager.

```
    public static function getRelations(): array
    {
        return [
            TenanciesRelationManager::class,
```

```
class TenanciesRelationManager extends RelationManager
{
    protected static string $relationship = 'tenancies';

    public function table(Table $table): Table
    {
        return TenancyResource::table($table)
            ->headerActions([
                CreateAction::make(),
            ]);
    }

    public function form(Form $form): Form
    {
        return TenancyResource::form($form);
    }
}
```

In that relation manager I return the TenancyResource form and table with the headerAction of CreateAction yet nothing appears at the top of the table as expected?

If I add the headerAction to the TenancyResource it works fine when viewing that table on it's own but it simply disappears on the relation. This happens for ALL my relation managers, none of them have any header actions.

![Screenshot 2023-07-25 at 13 50 08](https://github.com/filamentphp/filament/assets/18726355/6ee41090-13be-4577-9afc-c9c2a113a62b)

Am I doing something wrong?

**Comment by danharrin** [comment, marked as answer]
In v3, RM actions are hidden on the view page, and usable on the edit page. You can override this with the `isReadOnly()` method on the RM.

**Reply by Jam0r85** [reply]
This works! Brilliant, thank you, been racking my brain over this for the last few hours!

**Reply by bomshteyn** [reply]
@danharrin This default is very misleading imho the default RelationManager comes with Edit, Delete and Bulk Delete actions but they don't show up unless you add

```PHP
public function isReadOnly(): bool
{
    return false;
}
```

Wouldn't it make sense to have the default for the isReadOnly as false?

**Reply by danharrin** [reply]
1) That would be a breaking change
2) We implemented this due to opposite complaints in v2, about the actions being usable on the view page
3) It's 4 lines
4) It's in the documentation?

**Reply by bomshteyn** [reply]
1. Let's add isReadOnly into the stub
2. don't understand, if the actions are empty show if not don't
3. It's not about how many lines, it's that the stub as generated by `php artisan make:filament-relation-manager` makes no sense out of the box as it comes with CreateAction,  EditAction action and delete actions but they don't show in the view, which is very confusing. 
4. [this?](https://filamentphp.com/docs/3.x/panels/resources/relation-managers#read-only-mode) it's an end of a very long page the only way i saw it was after finding this discussion and searching for `isReadOnly` in docs after banging my head for the last half an hour. It's listed in the docs one before last after all the other edge cases...

So what i suggest is either adding to the `filament-relation-manager` stub

```PHP
public function isReadOnly(): bool
{
    return false;
}
```

or removing all Actions from the stub and adding doc's for RM Actions:
**Enabling Relation Actions**, not hiding it under **Read Only Mode**

**Reply by danharrin** [reply]
They do show though, they show on the Edit page of the resource. They just don't show on the View page? Which is meant to be a read-only page?

**Reply by danharrin** [reply]
It's not like the code is in the relation manager class and it's not being used anywhere, it's being used in every resource. Most people don't even have a View page

**Reply by danharrin** [reply]
My problem with putting it in the stub is that it's dead code for people who don't have a View page

**Reply by bomshteyn** [reply]
Ok, I guess i am in minority, I have view pages in 90% of my resources and don't see the issue of having an edit button / create button from there 🤷 

In many cases i only show RelationManager on the view page, not on the edit page as i feel it takes away from the edit focus.

**Reply by danharrin** [reply]
Right, well your default makes sense if you don't have RMs on the edit page. But since Filament's default is that there is no view page, and there is an edit page, and the edit page does have relation managers, I believe `isReadOnly()` has the correct implementation right now.

**Comment by daison12006013** [comment]
@Jam0r85, how did you add those switchable tabs?

Those (Tenancies / Media / Expenses / Certificates) did you modify your blade directly or there are ways to do it directly in filament without writing in blade file

**Reply by danharrin** [reply]
Just add more than one relation manager and it does it for you
